packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -195,11 +196,13 @@ const OuterApp = ({res, error, App, appState, routes, routerContext, location, d
     const AppConfig = getAppConfig()
     return (
         <Router location={location} context={routerContext}>
-            <DeviceContext.Provider value={{type: deviceType}}>
-                <AppConfig locals={res.locals}>
-                    <Switch error={error} appState={appState} routes={routes} App={App} />
-                </AppConfig>
-            </DeviceContext.Provider>
+            <CorrelationIdProvider correlationId={res.locals.requestId} resetOnPageChange={false}>
```

Comment:
This looks to be where you're using `res.locals` – you can avoid it, because you've got access to the response anyway!